src/PointwiseFunctions/AnalyticData/NewtonianEuler/KhInstability.hpp

```diff
@@ -0,0 +1,245 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/NewtonianEuler/Sources/NoSource.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticData/AnalyticData.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace NewtonianEuler {
+namespace AnalyticData {
+
+/*!
+ * \brief Initial data to simulate the Kelvin-Helmholtz instability.
+ *
+ * For comparison purposes, this class implements the planar shear of
+ * \cite Schaal2015 The initial state consists of a horizontal strip of mass
+ * density \f$\rho_\text{in}\f$ moving with horizontal speed
+ * \f$v_{\text{in}}\f$. The rest of the fluid possesses mass density
+ * \f$\rho_\text{out}\f$, and its horizontal velocity is \f$v_{\text{out}}\f$,
+ * both constant. Mathematically,
+ *
+ * \f{align*}
+ * \rho(x, y) =
+ * \begin{cases}
+ * \rho_\text{in}, & \left|y - y_\text{mid}\right| < b/2\\
+ * \rho_\text{out}, & \text{otherwise},
+ * \end{cases}
+ * \f}
+ *
+ * and
+ *
+ * \f{align*}
+ * v_x(x, y) =
+ * \begin{cases}
+ * v_{\text{in}}, & \left|y - y_\text{mid}\right| < b/2\\
+ * v_{\text{out}}, & \text{otherwise},
+ * \end{cases}
+ * \f}
+ *
+ * where \f$b > 0\f$ is the thickness of the strip, and \f$y = y_\text{mid}\f$
+ * is its horizontal bimedian. The initial pressure is set equal to a constant,
+ * and the system is evolved assuming an ideal fluid of known adiabatic index.
+ * Finally, in order to excite the instability, the vertical velocity is
+ * initialized to a small perturbation
+ *
+ * \f{align*}
+ * v_y(x, y) = A\sin(4\pi x)
```

Comment:
It looks like the periodic nature of the perturbation would have implications for the "allowed" sizes of the domain. Can these restrictions be described in the documentation (in a new paragraph)?